src/boot/efi/stub.c

```diff
@@ -388,7 +467,11 @@ static EFI_STATUS load_addons(
         if (err != EFI_SUCCESS)
                 return log_error_status(err, "Unable to open root directory: %m");
 
-        err = load_addons_from_dir(root, prefix, &items, &n_items, &n_allocated);
+        err = load_addons_from_dir(loaded_image, root, prefix, &addons, &n_items, &n_allocated);
+        if (err != EFI_SUCCESS)
+                return err;
+
+        err = load_addons_from_efi(loaded_image, &addons, &n_items, &n_allocated);
```

Comment:
This function will be called twice, due to two calls of `load_addons` from the function `run`. One call for Global addons and a second for uki addons.
Solution would be to add bool in argument, and conditionally call `load_addons_from_efi` if argument to true.

Reply:
An idea to fix that: 

```diff
diff --git a/src/boot/efi/stub.c b/src/boot/efi/stub.c
index 1e7ead8b55..69b913ddb2 100644
--- a/src/boot/efi/stub.c
+++ b/src/boot/efi/stub.c
@@ -437,6 +437,7 @@ static void items_free(char16_t **items, size_t n_items) {
 static EFI_STATUS load_addons(
                 EFI_HANDLE stub_image,
                 EFI_LOADED_IMAGE_PROTOCOL *loaded_image,
+                bool load_from_efi,
                 const char16_t *prefix,
                 const char *uname,
                 char16_t **ret_cmdline,
@@ -481,9 +482,11 @@ static EFI_STATUS load_addons(
         if (err != EFI_SUCCESS)
                 return err;
 
-        err = load_addons_from_efi(loaded_image, &addons, &n_items, &n_allocated);
-        if (err != EFI_SUCCESS)
-                return err;
+        if (load_from_efi) {
+                err = load_addons_from_efi(loaded_image, &addons, &n_items, &n_allocated);
+                if (err != EFI_SUCCESS)
+                        return err;
+        }
 
         if (n_items == 0)
                 return EFI_SUCCESS; /* Empty directory */
@@ -634,6 +637,7 @@ static EFI_STATUS run(EFI_HANDLE image) {
         err = load_addons(
                         image,
                         loaded_image,
+                        true,
                         u"\\loader\\addons",
                         uname,
                         &cmdline_addons_global,
@@ -650,6 +654,7 @@ static EFI_STATUS run(EFI_HANDLE image) {
                 err = load_addons(
                                 image,
                                 loaded_image,
+                                false,
                                 dropin_dir,
                                 uname,
                                 &cmdline_addons_uki,
```